internal/state/relationship/capturer.go

```diff
@@ -63,27 +83,78 @@ func (c *CapturerImpl) Capture(obj client.Object) {
 				Name:      svcName,
 			}
 		}
+	case *v1beta1.Gateway:
+		var selectors []labels.Selector
+		for _, listener := range o.Spec.Listeners {
+			if selector := graph.GetAllowedRouteLabelSelector(listener); selector != nil {
+				convertedSelector, err := metav1.LabelSelectorAsSelector(selector)
+				if err == nil {
+					selectors = append(selectors, convertedSelector)
+				}
+			}
+		}
+
+		gatewayName := client.ObjectKeyFromObject(o)
+		if len(selectors) > 0 {
+			c.gatewayLabelSelectors[gatewayName] = selectors
+			for ns, cfg := range c.referencedNamespaces {
+				var gatewayMatches bool
+				for _, selector := range selectors {
+					if selector.Matches(labels.Set(cfg.labelMap)) {
+						gatewayMatches = true
+						cfg.match = true
+						cfg.gateways[gatewayName] = struct{}{}
+						break
+					}
+				}
+				if !gatewayMatches {
+					// gateway labels have changed, so remove the namespace relationship
+					c.removeGatewayReferenceFromNamespaces(gatewayName, referencedNamespaces{ns: cfg})
```

Comment:
I think we can just `delete(cfg.gateways, gatewayName)` here - because in all the other lines in this block ,we directly manipulate `cfg.gateways` without a helper method. 

// I think this suggestion makes more sense if `cfg.match is removed`  though